ext/odbc/php_odbc.c

```diff
@@ -2359,6 +2368,9 @@ PHP_FUNCTION(odbc_result_all)
 					}
 					if (rc == SQL_SUCCESS_WITH_INFO) {
 						PHPWRITE(buf, result->longreadlen);
+					} else if (rc != SQL_SUCCESS) {
+						efree(buf);
+						RETURN_FALSE;
```

Comment:
Just returning false here is odd, as you'll end up with a partially printed table. Maybe you'll want to integrate this with the rc == SQL_ERROR check a few lines further up?

Reply:
`odbc_sql_error()` will only provide useful error info for `SQL_ERROR`. Closing the table makes sense, though.